cwf/gateway/tools/uesim_cli/main.go

```diff
@@ -296,12 +302,37 @@ func createUeConfig(imsi string, auth_key []byte, seq_num uint64) (*protos.UECon
 	if err != nil {
 		return nil, fmt.Errorf("could not generate OPc for subscriber: %s: %s", imsi, err)
 	}
+	ues, err := uecfg.GetMap("subscribers")
+	if err == nil {
+		ue, res := ues[imsi].(map[interface{}]interface{})
+		if res {
+			msisdn, ok = ue["msisdn"].(string)
```

Comment:
all those parameters should not be grab here. Similar as imsi and auth key, they should be passed to this function. Or move all the UE parameters here. Otherwise we have the logic all scatter

Reply:
UE can be created in uesim from 2 different CLI commands add_ue and handleTrafficGenCmd so the logic is already scattered. We can refactor by making changes in the existing code with a risk of modifying stuff that is working.

Reply:
Posting a new diff with some refactor to make  the logic more clean.